### PR TITLE
48. [POST] Post

### PR DESCRIPTION
**Pull request by DLiSWE**
Created createPost endpoint ( controller, service, route, added to dto )

### CODE REVIEW

File: src/controllers/post.controller.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { NextFunction, Request, Response } from 'express';
+import { PostService } from '@/services';
+import { BasePostDto, CreatePostRequestDto, CreatePostResponseDto } from '@/dtos';
+
+class PostController {
+  public postService = new PostService();
```
Comment (raylu511) [inline src/controllers/post.controller.ts:6]: probably should make this a private member

File: src/controllers/post.controller.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  public createPost = async (req: Request, res: Response, next: NextFunction) => {
```
Comment (raylu511) [inline src/controllers/post.controller.ts:8]: I'm thinking we should return the whole post object instead of the id.

File: src/services/index.ts [hunk 3]
```diff
@@ -1 +1,2 @@
 export { default as AuthService } from './auth.service';
+export { default as PostService } from './post.service';
```
Comment (raylu511) [inline src/services/index.ts:2]: newline at end of file

**Review by ZohaibManzoor00 — APPROVED** [review]

File: src/controllers/post.controller.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import { RequestWithUser } from '@/interfaces';
+
+class PostController {
+  private postService = new PostService();
+
+  public createPost = async (req: RequestWithUser, res: Response, next: NextFunction) => {
+    try {
+      const userId = req.user.id
+      const postData: CreatePostRequestDto = req.body;
+      const post : BasePostDto = await this.postService.createPosts({...postData, userId:userId});
```
Comment (raylu511) [inline src/controllers/post.controller.ts:13]: nit: you can just do `({ ...postData, userId })`

File: src/services/auth.service.ts [hunk 5]
```diff
@@ -16,7 +16,7 @@ class AuthService extends Repository<UserEntity> {
 
   private createToken({ id }: RegisterUserResponseDto): TokenData {
     const dataStoredInToken: DataStoredInToken = { id };
-    const expiresIn = 100;
+    const expiresIn = 10;
```
Comment (raylu511) [inline src/services/auth.service.ts:19]: lets revert this back

**Review by ZohaibManzoor00 — APPROVED** [review]
Looks good 
nit: lets see if we can map data to entities more efficiently

**Review by Matt-Tan15 — APPROVED** [review]

**Pull request merged**